src/main.rs

```diff
@@ -33,33 +34,77 @@ struct Args {
     wikipedia_urls: Option<PathBuf>,
 }
 
-fn write(dir: impl AsRef<Path>, page: Page) -> anyhow::Result<()> {
-    let Some(qid) = page.main_entity.map(|e| e.identifier) else {
-        // TODO: handle and still write
-        bail!("Page in list but without wikidata qid: {:?} ({})", page.name, page.url);
+/// Determine the directory to write the article contents to, create it, and create any necessary symlinks to it.
+fn create_article_dir(base: impl AsRef<Path>, page: &Page) -> anyhow::Result<PathBuf> {
+    let Some(qid) = page.wikidata() else {
+        // TODO: Handle and write to wikipedia title directory
+        info!("Page without wikidata qid: {:?} ({})", page.name, page.url);
+        return Ok(Default::default());
     };
 
-    let mut filename = dir.as_ref().to_owned();
-    filename.push(qid);
+    // Otherwise use wikidata as main directory and symlink from wikipedia titles.
+
+    let wikidata_dir = qid.get_dir(base.as_ref().to_owned());
+    create_dir_all(&wikidata_dir)
+        .with_context(|| format!("creating wikidata directory {:?}", &wikidata_dir))?;
+
+    // Write symlinks to main directory.
+    // TODO: Only write redirects that we care about.
+    for title in page.titles() {
+        let title = match title {
+            Err(e) => {
+                warn!("Unable to parse title for page {:?}:  {:#}", page.name, e);
+                continue;
+            }
+            Ok(title) => title,
+        };
+
+        // NOTE: Some wikipedia titles have '/' in them.
```

Comment:
How are they processed in the generator?

Reply:
The generator only works with complete wikipedia urls (and wikidata QIDs). The OSM tags like `en:Article_Title` are converted to urls somewhere early in the OSM ingestion process. 
It [dumps the urls to a file](https://github.com/organicmaps/organicmaps/blob/acc7c0547db4285dd8841ae7f98811268e38d908/generator/wiki_url_dumper.cpp#L63) for the descriptions scraper, then when it adds them to the mwm files [it strips the protocol, appends the url to the base directory, and looks for language html files in the folder at that location.](https://github.com/organicmaps/organicmaps/blob/34bbdf6a2f077b3d629b3f17e8e05bd18a4e4110/generator/descriptions_section_builder.cpp#L142).
It doesn't do any special processing for articles with a slash in the title, they are just another subdirectory down. I'll update the diagram to show that.

Reply:
Good. Please don't forget that if something can be simplified or improved by changing the current generator approach, then it makes sense to do it.

Reply:
I'm working on a list of changes that would be helpful